runtime/drivers/warehouse.go

```diff
@@ -0,0 +1,26 @@
+package drivers
+
+import (
+	"context"
+	"net/url"
+)
+
+type Warehouse interface {
+	// QueryAsFiles downloads results into files and returns an iterator to iterate over them
+	QueryAsFiles(ctx context.Context, props map[string]any, opt *QueryOption) (FileIterator, error)
+	Export(ctx context.Context, props map[string]any, store ObjectStore, outputLocation string) (*ExportResult, error)
```

Comment:
Maybe I'm missing a detail, but the `Export` function here looks very close to the model executor capabilities. 

Can it be removed and implemented as an `objectStoreToSelf` executor on the warehouse?

Reply:
So we have these options : 
1. Implement `selfToObjectStore` (I think it should be called `selfToObjectStore` and not `objectStoreToSelf`). This has to be done for every warehouse.
2. Or implement one `warehouseToSelfExecutor` in `S3` and implement `Export` method on each of the `Warehouse` which is how it is done right now.
Both seems equivalent to me. 
I can implement 1 if we want to emphasise on the usage of `ModelExecutors`. What do you suggest ?
Also in that case I think we can also refactor `QueryAsFiles` to `selfToFile`.

Reply:
I think option 1 sounds good. My thinking is that it is a nice principle to have drivers/executors be a one-way relationship – for example, currently, warehouses are aware of S3 and GCS because they are handled in the `Export` function *and* S3/GCS are aware of warehouses because they have an executor for them. But if we make this refactor, then warehouses will be aware of S3/GCS in the executor, but S3/GCS will not be aware of warehouses at all.

Additionally, the `Export` implementation already looks very close to an executor so should be an easy change to make.

Regarding `QueryAsFiles`, yeah that might make sense, though let's keep that for a follow up if it makes sense.